packages/web/src/components/apps/use-logs.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+import { AppChannel } from '@/clients/websocket';
+import { PreviewStatusPayloadType } from '@srcbook/shared';
+
+export type LogMessage = {
+  type: 'npm_install_error' | 'vite_error'; // TODO: add more types like "warning" or "problem"
+  timestamp: Date;
+  contents: string;
+};
+
+export interface LogsContextValue {
+  logs: Array<LogMessage>;
+  clearLogs: () => void;
+  addError: (message: Omit<LogMessage, 'timestamp'>) => void;
+  unreadLogsCount: number;
+
+  open: boolean;
+  togglePane: () => void;
+}
```

Comment:
I added a new `LogsContext` to allow anywhere in the app to easily have access to add new errors to this new panel. Here's what is in this context!